OperationDispatcher.getClusterInfo can block

### Describe the bug

Some tests may fail with the following trace:

```
java.lang.AssertionError: Blocking call! jdk.internal.misc.Unsafe#park on thread Thread[#9190,HotRod-client-async-pool-1431-5,5,ISPN-non-blocking-group]
at org.infinispan.util.CoreTestBlockHoundIntegration.lambda$applyTo$0(CoreTestBlockHoundIntegration.java:51)
at reactor.blockhound.BlockHound$Builder.lambda$install$8(BlockHound.java:488)
at reactor.blockhound.BlockHoundRuntime.checkBlocking(BlockHoundRuntime.java:89)
at java.base/jdk.internal.misc.Unsafe.park(Unsafe.java)
at java.base/java.util.concurrent.locks.LockSupport.park(LockSupport.java:221)
at java.base/java.util.concurrent.locks.StampedLock.acquireRead(StampedLock.java:1378)
at java.base/java.util.concurrent.locks.StampedLock.readLock(StampedLock.java:554)
at org.infinispan.client.hotrod.impl.transport.netty.OperationDispatcher.getClusterInfo(OperationDispatcher.java:185)
at org.infinispan.client.hotrod.impl.transport.netty.OperationDispatcher.getClientIntelligence(OperationDispatcher.java:317)
at org.infinispan.client.hotrod.impl.transport.netty.HeaderDecoder.readNewTopologyAndHash(HeaderDecoder.java:352)
at org.infinispan.client.hotrod.impl.transport.netty.HeaderDecoder.decode(HeaderDecoder.java:270)
at org.infinispan.client.hotrod.impl.transport.netty.HintedReplayingDecoder.callDecode(HintedReplayingDecoder.java:94)
at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:290)
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
at org.infinispan.client.hotrod.impl.transport.netty.IdleStateHandlerNoUnvoid.channelRead(IdleStateHandlerNoUnvoid.java:285)
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1357)
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:868)
at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:799)
at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:501)
at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:399)
at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:998)
at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
at java.base/java.lang.Thread.run(Thread.java:1583)
```
We need to add an exception to the topology lock as it held very briefly and why this error doesn't show up that often.

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Infinispan version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/persistence/remote/src/main/java/org/infinispan/persistence/remote/internal/RemoteStoreBlockHoundIntegration.java
+++ b/persistence/remote/src/main/java/org/infinispan/persistence/remote/internal/RemoteStoreBlockHoundIntegration.java
@@ -14,6 +14,7 @@ public class RemoteStoreBlockHoundIntegration implements BlockHoundIntegration {
       builder.allowBlockingCallsInside("org.infinispan.client.hotrod.impl.transport.netty.ChannelInitializer", "initSsl");
       // Some tests have client failover and it is possible for it need to acquire the read lock
       builder.allowBlockingCallsInside("org.infinispan.client.hotrod.impl.transport.netty.OperationDispatcher", "getCacheInfo");
+      builder.allowBlockingCallsInside("org.infinispan.client.hotrod.impl.transport.netty.OperationDispatcher", "getClusterInfo");
 
       // When a server fails we need to acquire the write lock very briefly to update internals
       builder.allowBlockingCallsInside("org.infinispan.client.hotrod.impl.transport.netty.OperationDispatcher", "handleConnectionFailure");

